app/Models/Contact/Contact.php

```diff
@@ -1522,6 +1522,16 @@ public function setStayInTouchTriggerDate($frequency, $triggerDate = null)
         $this->timestamps = $timestamps;
     }
 
+    /**
+     * Reset the notification date from current time
+     *
+     * @return void
+     */
+    public function resetStayInTouchTriggerDate()
+    {
+        $this->setStayInTouchTriggerDate($this->stay_in_touch_frequency ?? 0);
```

Comment:
Am I reading this right that the trigger gets reset from the time it was recorded? Should it instead be based off of when the event occurred? I'm thinking this could cause issues when retroactively filling out activities

Reply:
If a user creates a new event, but sets the date of that event in the past, the creation time isn't exactly the same as when they last interacted with the person.

For example, let's say on day 0 I set up a reminder for person X every 7 days. Then on day 6, I retroactively record an event with person X that happened on day 3. I would think I should be reminded about person X on day 10 (7 days after I last interacted with person X on day 3), but instead I'm reminded on day 13 (7 days after I recorded the event on day 6).

Maybe this is preferred since the reminder is based off of when you last thought about the person. Just calling it out as it's not how I originally interpreted the issue.

Reply:
I agree with @BobChubby . My intuition would be that if I added an event that happened a few days before the entry date, Monica would move the reminder date relative to the date of the event, rather than the creation of the event.